core/src/main/scala/com/salesforce/op/stages/impl/feature/SmartTextVectorizer.scala

```diff
@@ -77,6 +93,120 @@ class SmartTextVectorizer[T <: Text](uid: String = UID[SmartTextVectorizer[T]])(
     hashSpaceStrategy = getHashSpaceStrategy
   )
 
+  /* CODE FOR DETECTING SENSITIVE FEATURES BEGIN */
+  var guardCheckResults: Option[Array[Boolean]] = None
+
+  override def fit(dataset: Dataset[_]): SequenceModel[T, OPVector] = {
+    /* Set instance variable for guardCheck results here.
+    We compute guardChecks here because all of the logical checks can be implemented efficiently in native Spark,
+    and we would like to use the individual Text columns before they are combined in SequenceEstimator.fit().
+    It is not directly possible to do this computation in the treeAggregate() call below in fitFn() because one of the
+    logical checks computes a standard deviation, which requires knowing the mean beforehand. We could implement the
+    functionality in treeAggregate() if we could replace the standard deviation computation. */
+    if (getRemoveSensitive) {
+      guardCheckResults = Some(
+        inN.map(feature => guardChecks(dataset.asInstanceOf[Dataset[T#Value]], col(feature.name)))
+      )
+    }
+    // then call super
+    super.fit(dataset)
+  }
+
+  private def detectSensitive(dataset: Dataset[Seq[T#Value]]): NameIdentificationResults = {
+    val spark = dataset.sparkSession
+    val broadcastNameDict: Broadcast[NameDictionary] = spark.sparkContext.broadcast(NameDictionary())
+    val broadcastGenderDict: Broadcast[GenderDictionary] = spark.sparkContext.broadcast(GenderDictionary())
+
+    val MAX_NUM_NAMES = 50
+    def aggregateTwoResults(
+      one: NameIdentificationAccumulator, two: NameIdentificationAccumulator
+    ): NameIdentificationAccumulator = {
+      NameIdentificationAccumulator(
+        one.count + two.count,
+        one.predictedNameProb + two.predictedNameProb,
+        one.tokenInFirstNameDictionary + two.tokenInFirstNameDictionary,
+        one.tokenIsMale + two.tokenIsMale,
+        one.tokenIsFemale + two.tokenIsFemale,
+        one.tokenIsOther + two.tokenIsOther,
+        if (one.sampleOfNames.length + two.sampleOfNames.length <= MAX_NUM_NAMES) {
+          one.sampleOfNames ++ two.sampleOfNames
+        } else Random.shuffle(one.sampleOfNames ++ two.sampleOfNames).take(MAX_NUM_NAMES)
+      )
+    }
+
+    def aggregateSeqResults(
+      one: Seq[NameIdentificationAccumulator], two: Seq[NameIdentificationAccumulator]
+    ): Seq[NameIdentificationAccumulator] = {
+      one zip two map { case (x, y) => aggregateTwoResults(x, y) }
+    }
+
+    import com.salesforce.op.features.types.NameStats.GenderStrings._
+    def computeResults(input: T#Value): NameIdentificationAccumulator = {
```

Comment:
lets set up some time to talk about doing this in the fitFn with monoid aggregators